search-service/src/main/kotlin/com/egm/stellio/search/csr/model/ContextSourceRegistration.kt

```diff
@@ -0,0 +1,172 @@
+            )
+
+        fun compact(contexts: List<String>): RegistrationInfo =
+            RegistrationInfo(
+                entities = entities?.map { it.compact(contexts) },
+                propertyNames = propertyNames?.map { compactTerm(it, contexts) },
+                relationshipNames = relationshipNames?.map { compactTerm(it, contexts) }
+            )
+
+        fun validate(): Either<BadRequestDataException, Unit> = either {
+            return if (entities != null || propertyNames != null || relationshipNames != null) {
+                entities?.forEach { it.validate().bind() }
+                Unit.right()
+            } else {
+                BadRequestDataException("RegistrationInfo should have at least one element").left()
+            }
+        }
+    }
+
+    data class EntityInfo(
+        val id: URI? = null,
+        val idPattern: String? = null,
+        @JsonFormat(
+            with = [
+                JsonFormat.Feature.ACCEPT_SINGLE_VALUE_AS_ARRAY,
+                JsonFormat.Feature.WRITE_SINGLE_ELEM_ARRAYS_UNWRAPPED
+            ]
+        )
+        val type: List<String>
+    ) {
+        fun expand(contexts: List<String>): EntityInfo =
+            this.copy(
+                type = type.map { expandJsonLdTerm(it, contexts) },
+            )
+
+        fun compact(contexts: List<String>): EntityInfo =
+            this.copy(
+                type = type.map { compactTerm(it, contexts) },
+            )
+
+        fun validate(): Either<BadRequestDataException, Unit> {
+            val result = runCatching {
+                idPattern?.let { Pattern.compile(it) }
+                true
+            }.fold({ true }, { false })
+
+            return if (result)
+                Unit.right()
+            else BadRequestDataException("Invalid idPattern found in contextSourceRegistration").left()
+        }
+    }
+    fun expand(contexts: List<String>): ContextSourceRegistration =
+        this.copy(
+            information = information.map { it.expand(contexts) }
+        )
+
+    fun compact(contexts: List<String>): ContextSourceRegistration =
+        this.copy(
+            information = information.map { it.compact(contexts) }
+        )
+
+    fun serialize(
+        contexts: List<String>,
+        mediaType: MediaType = JSON_LD_MEDIA_TYPE,
+        includeSysAttrs: Boolean = false
+    ): String {
+        val representation = NgsiLdDataRepresentation(
+            entityRepresentation = EntityRepresentation.forMediaType(mediaType),
+            includeSysAttrs = includeSysAttrs,
+            attributeRepresentation = AttributeRepresentation.NORMALIZED
+        )
+        return dataTypeMapper.writeValueAsString(
+            dataTypeMapper.convertValue<Map<String, Any>>(
+                (this.compact(contexts))
+            ).plus(JSONLD_CONTEXT to contexts)
+                .toFinalRepresentation(representation) // todo dangerous?
```

Comment:
at least confusing. you could have a simple similar function in this class. that would also remove the need to use `NgsiLdDataRepresentation` which is really meant for entities.

Reply:
the toFinalRepresentation i will implement. Will be the same code as the one in Subscription.
I propose to regroup them in a DataTypeUtils in the shared library. (Is it ok for you?)

Reply:
Alongside the dataTypeMapper and the deserializeDataTypeAs